plugins/movetoarchive.koplugin/main.lua

```diff
@@ -29,6 +30,14 @@ function MoveToArchive:init()
     self.last_copied_from_dir = self.settings:readSetting("last_copied_from_dir")
 end
 
+-- check if the folder exists
+local function dir_exists_v1(path)
+  if (lfs.attributes(path, "mode") == "directory") then
+    return true
+  end
+  return false
+end
+
```

Comment:
Probably deserves its place around `util.pathExists` (which will also point out a much simpler way to write that ;)).

Reply:
I knew it I should have checked util first... :D Thanks!

Reply:
So, I should use this instead? Or I just move this function next to it? Because `util.pathExists doesn't care if it's file or directory, whereas I specifically need directory check.
https://github.com/koreader/koreader/blob/b949d07f4fbf7ffb0a145684f98f196bb0492aea/frontend/util.lua#L754-L759

Reply:
Something like `util.isDirectory()` next to it then. :-)

Reply:
Although I'd probably be fine with a simple `if lfs.attributes(path, "mode") == "directory" then` as well.

Reply:
Yeah, inlining it would be perfectly valid here (we probably do that already in the few places where we really want a *directory* instead of *something*).

Reply:
At the very least we already do it exactly that way in:

```lua
function util.makePath(path)
    if lfs.attributes(path, "mode") == "directory" then
        return true
    end
```

Reply:
> Although I'd probably be fine with a simple `if lfs.attributes(path, "mode") == "directory" then` as well.

So, without even making a new function in util?

Reply:
Yes, because in this case lfs is the "util". I don't think I would bother but either way.